File: P1/ejercicio4/PasswordLengthFilter.java
package ejercicio4;

public class PasswordLengthFilter implements Filter {
    private static final int MIN_LENGTH = 5;
    private static final int MAX_LENGTH = 50;

    @Override
    public void execute(Message message) throws Exception {
        String password = message.getPassword();
        if (password.length() < MIN_LENGTH || password.length() > MAX_LENGTH) {
            throw new Exception("Contraseña inválida: longitud incorrecta");
        }
    }
}

File: P2/email_filter_app/lib/filters/password_length_filter.dart
import 'package:email_filter_app/credentials.dart';
import 'package:email_filter_app/filters/filter.dart';

class PasswordLengthFilter extends Filter {
  static const int minLength = 5;
  static const int maxLength = 50;

  PasswordLengthFilter() : super("Validar longitud de contraseña");

  @override
  void execute(Credentials credentials) {
    final password = credentials.password;
    if (password.length < minLength || password.length > maxLength) {
      throw Exception('Contraseña inválida: longitud incorrecta');
    }
  }
}
